hr_period/models/hr_payslip_run.py

```diff
@@ -0,0 +1,192 @@
+                    raise UserError("The company on the selected period must "
+                                    "be the same as the company on the  "
+                                    "payslip batch.")
+        return True
+
+    @api.multi
+    @api.constrains('hr_period_id', 'schedule_pay')
+    def _check_period_schedule(self):
+        for run in self:
+            if run.hr_period_id:
+                if run.hr_period_id.schedule_pay != run.schedule_pay:
+                    raise UserError("The schedule on the selected period must "
+                                    "be the same as the schedule on the "
+                                    "payslip batch.")
+        return True
+
+    @api.model
+    def get_default_schedule(self, company_id):
+        company = self.env['res.company'].browse(company_id)
+
+        fy_obj = self.env['hr.fiscalyear']
+
+        fys = fy_obj.search([
+            ('state', '=', 'open'),
+            ('company_id', '=', company.id),
+        ])
+
+        return (
+            fys[0].schedule_pay
+            if len(fys) else 'monthly'
+        )
+
+    @api.onchange('company_id', 'schedule_pay')
+    @api.one
+    def onchange_company_id(self):
+        schedule_pay = self.schedule_pay or \
+            self.get_default_schedule(self.company_id.id)
+
+        if len(self.company_id) and schedule_pay:
+            period = self.env['hr.period'].get_next_period(self.company_id.id,
+                                                           schedule_pay,)
+            self.hr_period_id = period.id if period else False,
+
+    @api.onchange('hr_period_id')
+    def onchange_period_id(self):
+        if len(self.hr_period_id):
+            self.date_start = self.hr_period_id.date_start
+            self.date_end = self.hr_period_id.date_stop
+            self.date_payment = self.hr_period_id.date_payment
+            self.schedule_pay = self.hr_period_id.schedule_pay
+
+    @api.model
+    def create(self, vals):
+        """
+        Keep compatibility between modules
+        """
+        if vals.get('date_end') and not vals.get('date_payment'):
+            vals.update({'date_payment': vals['date_end']})
+        return super(HrPayslipRun, self).create(vals)
+
+    @api.multi
+    def get_payslip_employees_wizard(self):
+        """ Replace the static action used to call the wizard
+        """
+        self.ensure_one()
+        payslip_run = self[0]
+
+        view_ref = self.env['ir.model.data'].get_object_reference(
+            'hr_payroll', 'view_hr_payslip_by_employees')
+
+        view_id = view_ref and view_ref[1] or False
+
+        company = payslip_run.company_id
+
+        employee_obj = self.env['hr.employee']
+
+        employees = employee_obj.search([('company_id', '=', company.id)])
+
+        employee_ids = [
+            emp.id for emp in employees
+            if emp.contract_id.schedule_pay == payslip_run.schedule_pay
+        ]
+
```

Comment:
you can reduce these expressions in one:

employee_ids = self.env['hr.employee'].search([...]).ids